### DISCUSSION TITLE
How can I reuse a trained model to make future predictions in another script?

### DISCUSSION THREAD

**Discussion by gianluca-coletta** [autogluon/autogluon#3951] [Q&A] [answered]
Hi all, after building a 'TimeSeriesPredictor' object in the following way: 

```
predictor = TimeSeriesPredictor(
    prediction_length=12,
    freq='W-MON',
    target='target',
    path='weekly_predictor',
    eval_metric='MASE'
)
```
And after making the fit of the object 'TimeSeriesPredictor' in the following way:

```
predictor.fit(
    train_data,
    presets='best_quality',
    num_val_windows=5,
    time_limit=2400,
    random_seed=42
)
```
How can I reuse the predictor in another script to make predictions? Is there a save/load option?  Because I would like to make predictions using the best model without training again. I haven't found any documentation about this.

Also is it possible to see the hyperparameters of the model after doing HPO? 

Thank you all for the help!

**Comment by shchur** [comment, marked as answer]
Hi, you can save the predictor to a folder
```python
predictor = TimeSeriesPredictor(..., path="ag_predictor").fit(...)
```
and then load it from disk in a new script
```python
loaded_predictor = TimeSeriesPredictor.load("ag_predictor")
```

------

Investigating the model parameters is currently not a part of the public API, but you can do it as follows
1. Check the performance of each model with 
```python
predictor = TimeSeriesPredictor(...).fit(...)
predictor.leaderboard()
```
2. Load the best model with
```python
model = predictor._trainer.load_model("DeepAR/b96a619c")  # provide the model name here
model.get_user_params()
```

**Reply by gianluca-coletta** [reply]
Thank you @shchur for answering me so quickly

**Reply by gianluca-coletta** [reply]
Hi @shchur, sorry but this code doesn't work:

```
model = predictor._trainer.load_model("DeepAR/b96a619c")  # provide the model name here
model.get_user_params()
```

This is my code:

```
model = predictor._trainer.load_model("TemporalFusionTransformer/models/TemporalFusionTransformer/9ae2dec6")  # provide the model name here
model.get_user_params()
```

I try with this too:

```
model = predictor._trainer.load_model("TemporalFusionTransformer/9ae2dec6")  # provide the model name here
model.get_user_params()
```

But i get this error back:

`KeyError: 'TemporalFusionTransformer/9ae2dec6'`

And this is my predictor path

![image](https://github.com/autogluon/autogluon/assets/74981475/941a3ee0-dd1e-4a07-91a0-6d79614f89bc)

what am I doing wrong?

**Reply by shchur** [reply]
You can call `trainer.get_model_names()` to see the available model names.

**Reply by gianluca-coletta** [reply]
Ok, thanks @shchur . Seeing the model names I was able to solve the problem and use the code you suggested to see the hyperparameters of the models (there was a problem with the model name path)

Solution:

```
model = predictor._trainer.load_model("TemporalFusionTransformer\\eb7eb3c8")  
model.get_user_params()
```
